upstairs/src/lib.rs

```diff
@@ -5057,6 +5205,23 @@ async fn up_listen(
     println!("Wait for all three downstairs to come online");
     let mut lastcast = 1;
 
+    /*
+     * If this guest was configured with an IOPs limit, one branch of the
+     * loop below has to leak tokens. Leak every IOP_LEAK_MS
+     * milliseconds.
+     */
+    let guest_has_iop_limit = up.guest.get_iop_limit().is_some();
```

Comment:
If `guest_has_iop_limit` is set before the loop starts, it means the guest must set iops before starting up_main, otherwise the change might not be noticed?

I think it would work for an update to the amount, but a switch from `off` to `on` won't be notices, and I'm not sure what happens if we go from `on` to `off`

Reply:
> it means the guest must set iops before starting up_main, otherwise the change might not be noticed?

Yeah, you're right - this code does not support changing (or removing) the IOPs limit after the Upstairs starts.

Reply:
Note 0bad8cf changes the computation to be inside the check, so we could potentially change the limit(s) every 1 second.